runtime/libia2/init.c

```diff
@@ -12,6 +13,8 @@
 extern __thread void *ia2_stackptr_0[PAGE_SIZE / sizeof(void *)]
     __attribute__((aligned(4096)));
 
+uintptr_t ia2_stacks[16] IA2_SHARED_DATA = {0};
+
```

Comment:
Because this is `IA2_SHARED_DATA`, this is definitely not secure. That's fine if it's just for logging in permissive mode, but we should have a comment to that effect so nothing security-critical starts relying on it.

Reply:
Moved to #554.